server/Arcadia.Assistant.UserPreferences/UserPreferencesActor.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace Arcadia.Assistant.UserPreferences
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Akka.Actor;
+    using Akka.Persistence;
+
+    public class UserPreferencesActor : UntypedPersistentActor
+    {
+        private readonly Dictionary<string, UserPreferences> userPreferencesById = new Dictionary<string, UserPreferences>();
+
+        public override string PersistenceId => "user-preferences";
+
+        protected override void OnCommand(object message)
+        {
+            switch (message)
+            {
+                case GetUserPreferencesMessage msg:
+                    var userPreferences = this.userPreferencesById.ContainsKey(msg.UserId)
+                        ? this.userPreferencesById[msg.UserId]
+                        : null;
```

Comment:
what about creating and returning default preferences here? And sending SaveUserPreferencesMessages to Self to store it

Reply:
I use default preferences in user preferences service in Web project. But you're right, it is better to implement it here to allow other actors request for it.

Reply:
Done.

Reply:
Done, but I thought - why do we need to store default value into events? It should always be default if user didn't change them manually. But if we store default preferences to events, they will override if later we change some default behavior (e.g., make notifications disabled by default).